src/coreclr/vm/finalizerthread.cpp

```diff
@@ -569,6 +569,16 @@ void FinalizerThread::FinalizerThreadWait()
 
         EnableFinalization();
 
-        hEventFinalizerDone->Wait(INFINITE,TRUE);
+        // Under GC stress the finalizer queue may never go empty as frequent
+        // GCs will keep filling up the queue with items.
+        // We will disable GC stress to make sure the current thread is not permanently blocked on that.
+        GCStressPolicy::InhibitHolder iholder;
+
+        //----------------------------------------------------
+        // Do appropriate wait and pump messages if necessary
+        //----------------------------------------------------
+
+        DWORD status = hEventFinalizerDone->Wait(INFINITE,TRUE);
+        _ASSERTE(status == WAIT_OBJECT_0);
```

Comment:
This place does not complain about unused local.

Reply:
I am not sure I have before that this can break the build.

Reply:
Maybe the other case is included somewhere that is built with more strict rules.

Reply:
My guess is that this does not complain because `DWORD` is typedef, so it is a potentially complex type where the local variable may be impactful ... until some compiler decides to warn for this case too.

I would not hurt to change this one to the make-compiler-happy pattern.